python-package/xgboost/core.py

```diff
@@ -1339,6 +1369,20 @@ def _init(self, data: DataType, enable_categorical: bool, **meta: Any) -> None:
         self.handle = handle
 
 
+class DeviceQuantileDMatrix(QuantileDMatrix):
```

Comment:
Is this completely deprecated? Does that mean QuantileDMatrix automatically uses the device for device inputs?

Reply:
Yes.  The c++ layer does the dispatching.